lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,33 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) where n is the length of matrix
+# Space complexity: If a matrix is rows*columns large, than space complexity will be O(rows) or O(columns), depending on which is larger.
```

Comment:
If you look at the problem description, the number of rows and columns are the same.  It's a square matrix.